src/components/NavLink/index.tsx

```
@@ -15,13 +14,14 @@ export default function NavLink({
   href,
   ...props
-}: NavLinkProps) {
+}: NavLinkProps): ReactNode {
   const { asPath } = useRouter();
   const matches = exact ? asPath === href : asPath.startsWith(href.toString());
   const actualClassName = matches ? activeClassName : '';
 
   return (
     <Link href={href} {...props}>
+      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
       <a className={actualClassName}>{children}</a>
```

Biome (2.1.2)

[error] Provide a href attribute for the a element.

An anchor element should always have a href
Check this thorough explanation to better understand the context.

(lint/a11y/useValidAnchor)